### ISSUE TITLE
Yum failure prevented pbench from registering a tool

### ISSUE THREAD

**Issue by akrzos** [distributed-system-analysis/pbench#112]
This didn't cause any casualties until one of my machines added a repo while automation was running and some of my pbench output only has the vmstat tool rather than all of the tools I was expecting.

This is what prevented the tools (This case mpstat) from registering:

```
# register-tool --name=mpstat -- --interval=1
https://cdn.redhat.com/content/dist/rhel/server/7/7Server/x86_64/sjis/os/repodata/repomd.xml: [Errno 14] HTTPS Error 404 - Not Found
Trying other mirror.


 One of the configured repositories failed (Red Hat Enterprise Linux for S-JIS (RHEL 7 Server) (RPMs)),
 and yum doesn't have enough cached data to continue. At this point the only
 safe thing yum can do is fail. There are a few ways to work "fix" this:

     1. Contact the upstream for the repository and get them to fix the problem.

     2. Reconfigure the baseurl/etc. for the repository, to point to a working
        upstream. This is most often useful if you are using a newer
        distribution release than is supported by the repository (and the
        packages for the previous distribution release still work).

     3. Disable the repository, so yum won't use it by default. Yum will then
        just ignore the repository until you permanently enable it again or use
        --enablerepo for temporary usage:

            yum-config-manager --disable rhel-sjis-for-rhel-7-server-rpms

     4. Configure the failing repository to be skipped, if it is unavailable.
        Note that yum will try to contact the repo. when it runs most commands,
        so will have to try and fail each time (and thus. yum will be be much
        slower). If it is a very temporary problem though, this is often a nice
        compromise:

            yum-config-manager --save --setopt=rhel-sjis-for-rhel-7-server-rpms.skip_if_unavailable=true

failure: repodata/repomd.xml from rhel-sjis-for-rhel-7-server-rpms: [Errno 256] No more mirrors to try.
https://cdn.redhat.com/content/dist/rhel/server/7/7Server/x86_64/sjis/os/repodata/repomd.xml: [Errno 14] HTTPS Error 404 - Not Found
For some reason this tool could not be installed
```

To fix this I had to remove the broken repo. (Subscription-manager repos --disable=$bad_repo)  Unfortunately it is unclear as to why that repo decided to show up on one of my machines over the weekend.  All I could find related to that is this BZ comment: https://bugzilla.redhat.com/show_bug.cgi?id=1194899#c7

Prior to this failure, I have successful runs of pbench on the same machine with the other tools (mpstat and others.)  I am certain the tool was already installed and the yum failure should not affect pbench from registering that particular tool.

**Comment by atheurer** [comment]
On Mon, Nov 30, 2015 at 3:38 PM, Alex Krzos <notifications@github.com>
wrote:

> This didn't cause any casualties until one of my machines added a repo
> while automation was running and some of my pbench output only has the
> vmstat tool rather than all of the tools I was expecting.
>
> This is what prevented the tools (This case mpstat) from registering:
>
> # register-tool --name=mpstat -- --interval=1https://cdn.redhat.com/content/dist/rhel/server/7/7Server/x86_64/sjis/os/repodata/repomd.xml: [Errno 14] HTTPS Error 404 - Not Found
> Trying other mirror.
>
>
>  One of the configured repositories failed (Red Hat Enterprise Linux for S-JIS (RHEL 7 Server) (RPMs)),
>  and yum doesn't have enough cached data to continue. At this point the only
>  safe thing yum can do is fail. There are a few ways to work "fix" this:
>
>      1. Contact the upstream for the repository and get them to fix the problem.
>
>      2. Reconfigure the baseurl/etc. for the repository, to point to a working
>         upstream. This is most often useful if you are using a newer
>         distribution release than is supported by the repository (and the
>         packages for the previous distribution release still work).
>
>      3. Disable the repository, so yum won't use it by default. Yum will then
>         just ignore the repository until you permanently enable it again or use
>         --enablerepo for temporary usage:
>
>             yum-config-manager --disable rhel-sjis-for-rhel-7-server-rpms
>
>      4. Configure the failing repository to be skipped, if it is unavailable.
>         Note that yum will try to contact the repo. when it runs most commands,
>         so will have to try and fail each time (and thus. yum will be be much
>         slower). If it is a very temporary problem though, this is often a nice
>         compromise:
>
>             yum-config-manager --save --setopt=rhel-sjis-for-rhel-7-server-rpms.skip_if_unavailable=true
>
> failure: repodata/repomd.xml from rhel-sjis-for-rhel-7-server-rpms: [Errno 256] No more mirrors to try.https://cdn.redhat.com/content/dist/rhel/server/7/7Server/x86_64/sjis/os/repodata/repomd.xml: [Errno 14] HTTPS Error 404 - Not Found
> For some reason this tool could not be installed
>
> To fix this I had to remove the broken repo. (Subscription-manager repos
> --disable=$bad_repo) Unfortunately it is unclear as to why that repo
> decided to show up on one of my machines over the weekend. All I could find
> related to that is this BZ comment:
> https://bugzilla.redhat.com/show_bug.cgi?id=1194899#c7
>
> Prior to this failure, I have successful runs of pbench on the same
> machine with the other tools (mpstat and others.) I am certain the tool was
> already installed and the yum failure should not affect pbench from
> registering that particular tool.
>
​I think we have a problem with check_install_rpm() where the check for RPM
already installed always fails, then we run yum, and [before you
encountered this error] yum returns "already installed" and the problem
went unnoticed.  We won't fix the yum problem you saw, but we should fix
the install-check in pbench so you never call yum when you already have the
RPM.​

**Comment by akrzos** [comment]
$ver is not defined here:
[tool-scripts/sar - Line 150](https://github.com/distributed-system-analysis/pbench/blob/master/agent/tool-scripts/sar#L150)

I would fix it but I don't understand all of the code paths that adding a `ver="11.1.2"` variable will affect and if that will break other people's deployments of pbench.

**Comment by atheurer** [comment]
Ahh, this is totally broken.  We should not have any yum commands in the benchmark script itself.  I will try to fix later today.

**Comment by ndokos** [comment]
There is another yum command in kvmtrace and one in turbostat: those should probably go as well. pbench-sysstat is required by the specfile: do we want to do all of them that way?

**Comment by atheurer** [comment]
All scripts wanting a package should use check_install_rpm() from base file.  We actually should not require pbench_sysstat as a dependency (or any tools or benchmarks) for pbench-agent.

**Comment by atheurer** [comment]
FYI, we will probably change check_install_rpm to check_install_package and add support for different package managers within that function.  We might as well make the name change now, and fix all the current calls and replace the yum-installs with the new function name

**Comment by ndokos** [comment]
OK: I'll take pbench-sysstat off the specfile.

One problem with the above is that different distros name the packages differently, so it is not enough to change the function: we'll need to change the calls to it as well.

**Comment by atheurer** [comment]
This should be fixed with pull request #116
